src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -11139,36 +11165,27 @@ private bool ScanExplicitlyTypedLambda(Precedence precedence)
             var resetPoint = this.GetResetPoint();
             try
             {
-                bool foundParameterModifier = false;
-
                 // do we have the following, possibly with attributes before the parameter:
-                //   case 1: ( T x , ... ) =>
-                //   case 2: ( T x ) =>
-                //   case 3: ( out T x,
-                //   case 4: ( ref T x,
-                //   case 5: ( out T x ) =>
-                //   case 6: ( ref T x ) =>
-                //   case 7: ( in T x ) =>
+                //   case 1: ( T x [, ...]) =>
+                //   case 2: ( out T x [, ...]) =>
+                //   case 3: ( ref T x [, ...]) =>
+                //   case 4: ( in T x [, ...]) =>
                 //
                 // if so then parse it as a lambda
 
-                // Note: in the first two cases, we cannot distinguish a lambda from a tuple expression
-                // containing declaration expressions, so we scan forwards to the `=>` so we know for sure.
```

Comment:
trying to figure out what the full semantics are here now after your change.  Based on what i see above, it sounds like you're effectively saying: we *must* find a `=>` to be an explicitly typed lambda?  If so that is a bit of a pity.  wouldn't `(ref X y,` still be sufficient?  or was supporting that just took complex and we're rolling tha tback?  if so, that's fine, i just want to understand the reasoning here.

Reply:
ah, i see your comment below.  this makes sense.  note: i would intentionallyh add your comment below to the code.  that stuff really helps for maintenance.  i.e. that `F (ref T x` looks really confusing as either a lambda (which it is), or a possible invocation (which it is not... but which it is close to).

Reply:
Added comment.